[CC-1022] Updates

mafrosis opened the pull request:
Probably the only two contentious commits are:

 - 1861609 - renaming `application.py` to `manage.py`. This is primarily to remove ambiguity and (as I'm sure you know) the `manage.py` pattern is well know from the Django community.
 - 504f700 - renaming `app.py` to `config.py`, and then using `app.py` purely as the WSGI entrypoint. This a separation of concerns thing, and works well as a codebase scales.

https://artsalliancemedia.atlassian.net/browse/CC-1022

RangerOfFire commented on Makefile:
```diff
@@ -47,17 +47,17 @@ dependencies-dev: dependencies-python
 
 lint:
-	$(FLAKE8) --config=.flake8rc application
+	$(FLAKE8) --config=.flake8rc $(PROJECT)
```
Shouldn't this still be application as the folder in the repo root is still called application?

mafrosis replied:
The project var is defined at the top of the Makefile

mafrosis replied:
You are absolutely right, I was being damned fool

RangerOfFire commented on application/config.py:
```diff
@@ -0,0 +1,36 @@
+
+from flask import Flask
+
+from application.env_utils import load_config
```
It feels like this file has very little to do with config, if this is an accepted design pattern then ignore me.

mafrosis replied:
I had meant to merge `env_utils.py` into `config.py`.. Not sure how I missed that :) Update coming.

RangerOfFire commented on env_conf/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM python:2.7-slim
+FROM python:3.4-slim
```
:+1:

RangerOfFire commented on test/conftest.py:
```diff
@@ -0,0 +1,62 @@
+@pytest.yield_fixture(scope='session')
+def flask_app():
+    """
+    Provide test isolation by creating 
```
This docstring feels unfinished :racehorse:

mafrosis replied:
Good spot

RangerOfFire commented:
Structure looks good, just the minor things I commented on.

shipperizer commented:
looks good, I would just copy the uwsgi config as well and maybe the logging part from the user-service, but it's arguable, so up to you
here's my :sweet_potato:

mafrosis commented:
@shipperizer I thought we could add the logging stuff when we've finalised it. Maybe it's ready now?

shipperizer commented:
@mafrosis good to me, i was just lazy and didn't want to do it on my own :dart:

mafrosis commented:
@shipperizer @RangerOfFire 
Notable changes in recent iteration:
- renamed the `application` app to `flask_template`, to make it explicit and unambiguous
- merged `env_utils.py` and `config.py`, and removed unnecessarily complicated `load_env_var()` function
- included logging code and uwsgi config :)

shipperizer commented:
:sweet_potato:

lorantonodi commented:
👍

RangerOfFire commented on flask_template/config.py:
```diff
@@ -0,0 +1,80 @@
+
+    # load DEBUG env var and convert to boolean
+    config['DEBUG'] = os.environ.get('DEBUG', False)
+    if isinstance(config['DEBUG'], str) and config['DEBUG'].lower() == 'true':
```
I'm slightly iffy about the need for this but whatever it works.

mafrosis replied:
The previous implemented of this created a `value_is` lambda which is passed into `load_env_var` where it wraps `os.environ.get`:

https://github.com/artsalliancemedia/flask-template/blob/master/application/env_utils.py#L28

I thought that was a bit over-engineered. Hence the very simple alternative approach.

RangerOfFire commented:
:accept:
